### PR TITLE
Practice day3 shrasti

### PR DESCRIPTION
**Pull request by Shrasti0207**
Practice On Pattern Matching , Sealed Class and list

### CODE REVIEW

File: scala-practice/src/main/scala/ScalaPracticeDay3.sc [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    case personAge if personAge._2 < 18 => "underage"
+    case personAge if personAge._2 < 65 => "adult"
+    case _ => "senior"
+  }
+}
+logger.info("Person's age nature :" + describePerson("shrasti", 21))
+
+def evenOrOdd(number: Int): String = {
+  number match {
+    case num if num % 2 == 0 => s"$num is even"
```
Comment (muskan3006) [inline scala-practice/src/main/scala/ScalaPracticeDay3.sc:17]: num is not good name

File: scala-practice/src/main/scala/ScalaPracticeDay3.sc [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    case num if num % 2 != 0 => s"$num is odd"
+  }
+}
+logger.info("Find even or odd the number :" + evenOrOdd(3))
+logger.info("Find even or odd the number :" + evenOrOdd(-2))
+
+def flattenListusingFlatMap(inputList: List[List[Int]]): List[Int] = {
+  inputList.flatten(x => x)
```
Comment (muskan3006) [inline scala-practice/src/main/scala/ScalaPracticeDay3.sc:25]: what is x? and do we even require this expression(x => x) here? what else way we can do that?

File: scala-practice/src/main/scala/ScalaPracticeDay3.sc [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+val fruits = List("apple", "apple", "orange", "pear", "pear", "pear")
+logger.info("group by method : " + fruits.groupBy(identity))
+
+sealed abstract class Experiment
+
+case class Var(name: String) extends Experiment
+
+case class Number(num: Int) extends Experiment
+
+case class sum(num1: Int, num2: Int) extends Experiment
```
Comment (muskan3006) [inline scala-practice/src/main/scala/ScalaPracticeDay3.sc:48]: naming convention wrong

File: scala-practice/src/main/scala/ScalaPracticeDay3.sc [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+def describe(e: Experiment): String = e match {
```
Comment (muskan3006) [inline scala-practice/src/main/scala/ScalaPracticeDay3.sc:50]: naming convention

**Review by muskan3006 — CHANGES_REQUESTED** [review]
please resolve the comment and update PR with latest main

File: scala-practice/src/main/scala/ScalaPracticeDay3.sc (hunk 1 above)
Comment (Shrasti0207) [reply scala-practice/src/main/scala/ScalaPracticeDay3.sc]: I have solved num into matchNumber

File: scala-practice/src/main/scala/ScalaPracticeDay3.sc (hunk 2 above)
Comment (Shrasti0207) [reply scala-practice/src/main/scala/ScalaPracticeDay3.sc]: I have solved

File: scala-practice/src/main/scala/ScalaPracticeDay3.sc (hunk 3 above)
Comment (Shrasti0207) [reply scala-practice/src/main/scala/ScalaPracticeDay3.sc]: I have solved num1 into inputNumber1 and num2 into inputNumber2

File: scala-practice/src/main/scala/ScalaPracticeDay3.sc (hunk 4 above)
Comment (Shrasti0207) [reply scala-practice/src/main/scala/ScalaPracticeDay3.sc]: e updated into expression

**Review by muskan3006 — APPROVED** [review]

**Pull request merged**